File: Python/gcd.py
# Function to find HCF the Using Euclidian algorithm
def gcd(x, y):
   while(y):
       x, y = y, x % y
   return x

num1 = int(input())
num2 = int(input())

gcd = gcd(num1, num2)
print("The GCD is", gcd)

"""
4
6
The H.C.F. is 2

"""
"""
time complexity O(logn)
space complexity O(1)
"""

File: JavaScript/gcd.js
//function to find gcd of two numbers 
//using euclidean algorithm

function gcd(x, y) {
    let temp;
    if (x === 0) return y;
    else if (y === 0) return x;
    else {
        while (y) {
            temp = x;
            x = y;
            y = temp % y
        }
        return x;
    }
}

console.log(gcd(num1, num2));


/*
Input -> num1 = 17, num2 = 6
Output -> 1

Input -> num1 = 4, num2 = 6
Output -> 2

---Time complexity O(logn)
---Space complexity O(1)
*/